### DISCUSSION TITLE
Too many instances of Newtonsoft.Json

### DISCUSSION THREAD

**Discussion by JimmyCushnie** [starikcetin/awesome-upm#29] [Existing]
The table lists three different forks of Netwonsoft.Json. None of them are meaningfully different from each other.

Also, none of them are very popular. I propose they all get replaced with [this](https://github.com/jilleJr/Newtonsoft.Json-for-Unity) much more popular, much more active fork.

**Comment by MeikelLP** [comment]
There's also an "official" one provided by Unity.

**Comment by favoyang** [comment]
@MeikelLP I don't know it's a bug or something else, the *com.unity.nuget.newtonsoft-json* is hidden in the package manager ui by default. See the screenshot, the package list has no one starts with letter N (show preview is on).

![image](https://user-images.githubusercontent.com/125390/72681577-e9b13e00-3aff-11ea-9189-699062511959.png)

However, if I manually add the package name to dependency. Or use my [CLI](https://github.com/openupm/openupm-cli#openupm-cli):
```
openupm add com.unity.nuget.newtonsoft-json
added: com.unity.nuget.newtonsoft-json@2.0.0-preview
manifest updated, please open unity project to apply changes
```

It just appears.
![image](https://user-images.githubusercontent.com/125390/72681653-97bce800-3b00-11ea-9ff5-4c20429a5321.png)

Then it behaves normally in the all package list, even I remove the package. Anyway this is a bit off-topic.

**Comment by favoyang** [comment]
While, at least they have very different namespaces... After https://github.com/jilleJr/Newtonsoft.Json-for-Unity/issues/33, we can include https://github.com/jilleJr/Newtonsoft.Json-for-Unity to the awesome-upm. 

For now, I will either use the Unity one or https://github.com/PixelWizards/com.newtonsoft.json (current available on [OpenUPM](https://openupm.com/packages/com.newtonsoft.json/))

**Comment by MeikelLP** [comment]
I'm only using Unity's json package. AFAIK the package is hidden because it is in the  `staging` group: https://bintray.com/unity/unity-staging

I think only packages from https://bintray.com/unity/unity are listed in the UI (?)

**Comment by applejag** [comment]
Using said Newtonsoft.Json package from PixelWizards is highly discouraged as it is just using Unity's package (<https://bintray.com/unity/unity/com.unity.nuget.newtonsoft-json>) that's only meant for internal use at Unity, not for users of Unity. To quote the `/Documentation~/index.md` inside Unity's package itself:

> ***This is a package intended for internal Unity Development Projects and as such this package is not supported. Use at your own risk.***

More importantly is that you will not be able to get any support from Unity because of this.

And for the record, version 2.0.0-preview of their package is actually using the AOT build from my [npm released version 12.0.1 (old and outdated)](https://www.npmjs.com/package/jillejr.newtonsoft.json-for-unity/v/12.0.1).

I've in some mail chats with a Unity developer tried to reason about this, and it seems they will be wanting to take in external packages to be available through Unity without having to add custom scoped registries, but that plan is not ready yet.

With my repo I've tried to unify the lessons from the previous repos and their creators. ParentElement, author of the original JSON. NET assets store package is about to start helping with my fresh repo, and I'm hoping to get hold of SaladLabs soon too.

**Comment by MeikelLP** [comment]
For me personally I don't think I will ever need support for that package. And if Unity is using it internally I will use it as well because it is guaranteed to work.

Most of Unity's useful stuff is "internal" or not open for public. It is there but Unity doesn't allow you to use it in the conventional way. That is/was always annoying about Unity.

My opinion.

**Comment by applejag** [comment]
Understandable opinion but I highly disagree. It is not guaranteed to work, it's just that it works enough for the Unity team that uses said package. Seeing that Unity even snagged my version directly it's fare to say that my package is ok to use. There's even some bugs in the version that the Unity devs snagged that they haven't updated regarding IL2CPP builds and strong naming.

Sorry if I just jumped into the discussion. But there are not any benefits of using Unity's temp/internal package when there's more full fletched solutions available. Main difference is that mine requires a custom scoped registry, which is not a burden at all.

**Comment by starikcetin** [comment]
Alright, to wrap up:

* @k0dep and @Wolfoso repos has been removed due to legal reasons. ( https://github.com/starikcetin/awesome-upm/pull/32 )
* @jilleJr repo has been added ( https://github.com/starikcetin/awesome-upm/pull/31 )

In the end, we have 2 Newtonsoft.Json repos in total, which is acceptable in my opinion. I am closing this issue for now, but you can feel free to continue the discussion.
